### PR TITLE
Fix #753: Change platform into JPA enum

### LINKED ISSUE #753: Change platform into JPA enum
Change field `PushDeviceRegistrationEntity#platform` type of `String` into `Enum` which improves type safety. Mind that using `@Enumerated(EnumType.STRING)` is not powerful enough, lower-cases need to be kept because of backward compatibility.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: powerauth-push-server/src/main/java/io/getlime/push/repository/model/PlatformConverter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            case ANDROID -> "android";
+        };
+    }
+
+    @Override
+    public Platform convertToEntityAttribute(final String dbData) {
+        return switch (dbData) {
+            case "ios" -> Platform.IOS;
+            case "android" -> Platform.ANDROID;
+            default -> throw new MappingException("No mapping for platform: " + dbData);
```
Comment (romanstrobl) [inline powerauth-push-server/src/main/java/io/getlime/push/repository/model/PlatformConverter.java:44]: How is this exception being handled? Is it correct to use a Hibernate exception for a class which is not database related?

**Review by romanstrobl — APPROVED** [review]
This is a nice refactoring, however, let's review the error handling.

File: powerauth-push-server/src/main/java/io/getlime/push/repository/model/PlatformConverter.java (hunk 1 above)
Comment (banterCZ) [reply powerauth-push-server/src/main/java/io/getlime/push/repository/model/PlatformConverter.java:44]: Good question. The class `package private`, is placed in the package `io.getlime.push.repository.model`, so I consider `repository` as a database package, and used by the entity `PushDeviceRegistrationEntity`.

Comment (banterCZ) [reply powerauth-push-server/src/main/java/io/getlime/push/repository/model/PlatformConverter.java]: At the end, I have removed the Hibernate exception.

**Review by jandusil — APPROVED** [review]
Good addition

**Pull request merged**